typing/printtyp.mli

```diff
@@ -93,29 +93,37 @@ module Conflicts: sig
 end
 
 val reset: unit -> unit
-val mark_loops: type_expr -> unit
-val reset_and_mark_loops: type_expr -> unit
-val reset_and_mark_loops_list: type_expr list -> unit
 
+(** Print out a type.  This type expression will not share type variable names
+    with any other type expressions; this function resets the global printing
+    state first.  If you want multiple types to use common names for type
+    variables, see [prepare_for_printing] and [prepared_type_expr].  *)
```

Comment:
It might be clearer to mention that the function is not sharing names for common type variables. Something like

>  The printed type expression will not share names for common type variable with any other type expressions.
> If you want multiple types to use common names for type variables, see [prepare_for_printing] and [prepared_type_expr].

?

I am not sure how useful it is to mention a printing state that we are resetting.

Reply:
I like this idea; I went with

> Print out a type.  This will pick names for type variables, and will not reuse names for common type variables shared across multiple type expressions.  (It will also reset the printing state, which matters for other type formatters such as [prepared_type_expr].)  If you want multiple types to use common names for type variables, see [prepare_for_printing] and [prepared_type_expr].